ports/stm/common-hal/microcontroller/Pin.c

```diff
@@ -194,3 +194,12 @@ void common_hal_mcu_pin_claim(const mcu_pin_obj_t* pin) {
 void common_hal_mcu_pin_reset_number(uint8_t pin_no) {
     reset_pin_number(pin_no / 16, pin_no % 16);
 }
+
+void set_drive_mode(const mcu_pin_obj_t *pin, uint32_t mode) {
```

Comment:
This should be a static function within PWMAudioOut.c, it doesn't need to be in Pin.c. The function in DigitalIO is very old and shouldn't be adjusting the pull and speed as it does, so it needs to be tweaked anyway.